tests/test_trtllm_moe_allreduce_fusion.py

```diff
@@ -0,0 +1,410 @@
+
+                            # 3. All-Reduce
+                            all_reduced_ref = moe_out_ref.clone().to(dtype)
+                            dist.all_reduce(all_reduced_ref, group=group)
+                            all_reduced_ref = all_reduced_ref.to(torch.float32)
+
+                            # 4. Fused Ops
+                            ref_residual_out = all_reduced_ref + residual_in_clone.view(
+                                token_num, HIDDEN_SIZE
+                            ).to(torch.float32)
+
+                            variance = (
+                                ref_residual_out.to(torch.float32)
+                                .pow(2)
+                                .mean(dim=-1, keepdim=True)
+                            )
+                            hidden_states = ref_residual_out * torch.rsqrt(
+                                variance + rms_eps
+                            )
+                            ref_norm_out = rms_gamma.to(torch.float32) * hidden_states
+
+                            # 5. Run kernel
+                            torch.cuda.synchronize()
+
+                            comm.trtllm_moe_allreduce_fusion(
+                                world_size=world_size,
+                                world_rank=rank,
+                                token_num=token_num,
+                                hidden_dim=HIDDEN_SIZE,
+                                workspace_ptrs=workspace_tensor,
+                                launch_with_pdl=launch_with_pdl,
+                                residual_in=residual_in,
+                                rms_gamma=rms_gamma,
+                                rms_eps=rms_eps,
+                                scale_factor=scale_factor,
+                                moe_reduction_device_num_experts=active_expert_num,
+                                moe_reduction_scale_input=moe_reduction_scale_input,
+                                moe_reduction_active_experts_token_input=moe_reduction_active_experts_token_input,
+                                moe_reduction_token_input=moe_reduction_token_input,
+                                layout_code=swizzled_layout_code,
+                                allreduce_out=all_reduce_out,
+                                residual_out=residual_out,
+                                norm_out=norm_out,
+                                quant_out=quant_out,
+                                scale_out=scale_out,
+                            )
+
+                            # match shape
+                            all_reduce_out = all_reduce_out.view(token_num, HIDDEN_SIZE)
+                            residual_out = residual_out.view(token_num, HIDDEN_SIZE)
+                            norm_out = norm_out.view(token_num, HIDDEN_SIZE)
+
+                            torch.cuda.synchronize()
+
+                            # 6. Check correctness
+                            tolerance = 8e-2 if dtype == torch.float16 else 8e-1
```

Comment:
8e-1 seems too large for me, can you give an example about the distribution of `all_reduce_out`?